### PR TITLE
Fix a bug in the backprop tutorial

### LINKED ISSUE #1050: [BUG] Backprop tutorial has an incorrect gradient result compared to other built-in methods
Reported here: https://discuss.pennylane.ai/t/possible-discrepancy-in-the-quantum-gradients-with-backpropagation-tutorial/4236
Demo: https://pennylane.ai/qml/demos/tutorial_backprop/

The gradient calculated manually via the below doesn't match what `jax.grad` or `qml.gradients.param_shift` gives 🤔.

```
def parameter_shift_term(qnode, params, i):
    shifted = params.copy()
    shifted = shifted.at[i].add(np.pi/2)
    forward = qnode(shifted)  # forward evaluation

    shifted = shifted.at[i].add(-np.pi/2)
    backward = qnode(shifted) # backward evaluation

    return 0.5 * (forward - backward)
```

### PR DESCRIPTION
**Pull request by josh146**
The parameter shift rule implementation was not correctly specifying the backward shift.

Closes #1050

### CODE REVIEW

**Review by mankrit-singh — APPROVED** [review]
verified it on my Colab notebook

File: demonstrations/tutorial_backprop.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def parameter_shift_term(qnode, params, i):
     shifted = shifted.at[i].add(np.pi/2)
     forward = qnode(shifted)  # forward evaluation
 
-    shifted = shifted.at[i].add(-np.pi/2)
```
Comment (isaacdevlugt) [inline demonstrations/tutorial_backprop.py:115]: Ah! good catch 👌

**Review by isaacdevlugt — APPROVED** [review]

**Review by KetpuntoG — APPROVED** [review]
